tx-pool/src/component/verify_queue.rs

```diff
@@ -0,0 +1,154 @@
+}
+
+/// The verify queue is a priority queue of transactions to verify.
+pub struct VerifyQueue {
+    /// inner tx entry
+    inner: MultiIndexVerifyEntryMap,
+    /// used to notify the tx-pool to update the txs count
+    queue_tx: watch::Sender<usize>,
+}
+
+impl VerifyQueue {
+    /// Create a new VerifyQueue
+    pub(crate) fn new(queue_tx: watch::Sender<usize>) -> Self {
+        VerifyQueue {
+            inner: MultiIndexVerifyEntryMap::default(),
+            queue_tx,
+        }
+    }
+
+    /// Returns the number of txs in the queue.
+    pub fn len(&self) -> usize {
+        self.inner.len()
+    }
+
+    /// Returns true if the queue contains no txs.
+    #[allow(dead_code)]
+    pub fn is_empty(&self) -> bool {
+        self.inner.is_empty()
+    }
+
+    /// Returns true if the queue is full.
+    pub fn is_full(&self) -> bool {
+        self.len() > DEFAULT_MAX_VERIFY_TRANSACTIONS
+    }
+
+    /// Returns true if the queue contains a tx with the specified id.
+    pub fn contains_key(&self, id: &ProposalShortId) -> bool {
+        self.inner.get_by_id(id).is_some()
+    }
+
+    /// Shrink the capacity of the queue as much as possible.
+    pub fn shrink_to_fit(&mut self) {
+        shrink_to_fit!(self.inner, SHRINK_THRESHOLD);
+    }
+
+    /// Remove a tx from the queue
+    pub fn remove_tx(&mut self, id: &ProposalShortId) -> Option<Entry> {
+        self.inner.remove_by_id(id).map(|e| {
+            self.shrink_to_fit();
+            e.inner
+        })
+    }
+
+    /// Remove multiple txs from the queue
+    pub fn remove_txs(&mut self, ids: impl Iterator<Item = ProposalShortId>) {
+        for id in ids {
+            self.inner.remove_by_id(&id);
+        }
+        self.shrink_to_fit();
+    }
+
+    /// Returns the first entry in the queue and remove it
+    pub fn pop_first(&mut self) -> Option<Entry> {
+        if let Some(entry) = self.get_first() {
+            self.remove_tx(&entry.tx.proposal_short_id());
+            Some(entry)
+        } else {
+            None
+        }
+    }
+
+    /// Returns the first entry in the queue
+    pub fn get_first(&self) -> Option<Entry> {
+        self.inner
+            .iter_by_added_time()
+            .next()
+            .map(|entry| entry.inner.clone())
+    }
+
+    /// If the queue did not have this tx present, true is returned.
+    /// If the queue did have this tx present, false is returned.
+    pub fn add_tx(
+        &mut self,
+        tx: TransactionView,
+        remote: Option<(Cycle, PeerIndex)>,
+    ) -> Result<bool, Reject> {
+        if self.contains_key(&tx.proposal_short_id()) {
+            return Ok(false);
+        }
+        if self.is_full() {
+            return Err(Reject::Full(format!(
+                "chunk is full, tx_hash: {:#x}",
+                tx.hash()
+            )));
+        }
+        self.inner.insert(VerifyEntry {
+            id: tx.proposal_short_id(),
+            added_time: std::time::SystemTime::now()
```

Comment:
How about `ckb_systemtime::system_time_as_millis()`? It is extensively used in the ckb repo.